### PR TITLE
Fine-tune watchlist filtering

### PR DESCRIPTION
**Pull request by maciejhirsz**

### CODE REVIEW

File: backend/src/onfido.js [hunk 1]
```diff
@@ -341,21 +342,35 @@ function hashDocumentNumbers (documentNumbers) {
...
+    const sources = record.sources;
+    const recordDob = record.entity_fields_dob;
+
+    // Ignore PEPs lists
+    if (!sources.split(',').find((source) => !PEPS_PATTERN.test(source))) {
+      return false;
     }
+
+    // Filter out records with different date of birth
+    return !dob || recordDob === dob || recordDob === shortDob;
```
Comment (ngotchac) [inline backend/src/onfido.js]: `!dob` could be higher, right?

File: backend/src/onfido.js [hunk 2]
```diff
@@ -341,21 +342,35 @@ function hashDocumentNumbers (documentNumbers) {
...
-function verifyWatchlist (watchlistReport) {
-  const { breakdown } = watchlistReport;
+function verifyWatchlist (watchlistReport, dob) {
+  const shortDob = dob.substr(0, 4);
+  const { properties } = watchlistReport;
+  const { records = [] } = properties;
 
-  for (const category of REQUIRED_WATCHLIST_CATEGORIES) {
-    const { result } = breakdown[category];
+  const PEPS_PATTERN = /\bpeps?\b/i;
```
Comment (ngotchac) [inline backend/src/onfido.js]: This could be set in higher scope.

File: backend/src/onfido.js [hunk 3]
```diff
@@ -341,21 +342,35 @@ function hashDocumentNumbers (documentNumbers) {
...
  * @return {Object}
  */
-function verifyWatchlist (watchlistReport) {
-  const { breakdown } = watchlistReport;
+function verifyWatchlist (watchlistReport, dob) {
+  const shortDob = dob.substr(0, 4);
+  const { properties } = watchlistReport;
+  const { records = [] } = properties;
 
-  for (const category of REQUIRED_WATCHLIST_CATEGORIES) {
```
Comment (ngotchac) [inline backend/src/onfido.js]: `REQUIRED_WATCHLIST_CATEGORIES` isn't used anymore?

File: backend/src/onfido.js (hunk 1 above)
Comment (maciejhirsz) [reply backend/src/onfido.js]: Actually that should be `!recordDob` :see_no_evil:

Comment (maciejhirsz) [reply backend/src/onfido.js]: Basically, keep suspected records if:
+ Date of birth on the record matches the date of birth on the document
+ OR there is no date of birth on the record (no way to verify validity, so we stay cautious)

**Pull request merged**